### PR TITLE
fix: simplify plugins and eliminate enums

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
#### Simplify enums plugin development

Previously creating plugins require 4 properties/methods specifically `name`, `type`, `setup`, and `execute`. With this change, `name` defaults to a UUID, type defaults to `"enrichment"`, setup defaults to a no-op fn. for enrichment plugin, `execute` defaults to a passthrough fn. for destination plugin, `execute` is required. Below is the summary of changes:

| Field | Required? | Default |
|--------|--------|--------|
| name | No | UUID |
| type | No | `"enrichment"` | 
| setup | No | no-op | 
| execute | For destination plugin only | passthrough | 

```diff
const enrichmentPlugin: EnrichmentPlugin = {
-  // the values below are the new defaults. omitting will achieve similar results.
-   name: UUID(),
-   type: 'enrichment',
-   setup: async () => undefined,
-   execute: async (e) => e,
};
```

#### Eliminate enums

Removes enums for `TransportType` `ServerZone` and `PluginType`. Removes overhead of learning how to import and use enum. Replaces with old enum's literal values.

**Setting transport provider on initialization**

```diff
  import * as amplitude from '@amplitude/analytics-browser';

  amplitude.init(API_KEY, USER_ID, {
-   transport: amplitude.Types.TransportType.Fetch,
+   transport: 'fetch',
  });
```

**Setting transport provider using `setTransport()`**

```diff
  import * as amplitude from '@amplitude/analytics-browser';

- amplitude.setTransport(amplitude.Types.TransportProvider.Fetch);
+ amplitude.setTransport('fetch');
```

**Setting server zone on initialization**

```diff
  import * as amplitude from '@amplitude/analytics-browser';

  amplitude.init(API_KEY, USER_ID, {
-   serverZone: amplitude.Types.ServerZone.US,
+   serverZone: 'US',
  });
```

**Creating plugins**

```diff
import * as amplitude from '@amplitude/analytics-browser';

const plugin: EnrichmentPlugin = {
  name: 'my-plugin',
-   type: amplitude.Types.PluginType.ENRICHMENT;
+  type: 'enrichment'; 
  setup: async () => undefined;
  execute: async (e) => e;
};
```

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  Yes

### CODE REVIEW

File: packages/analytics-browser-test/test/index.test.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { default as nock } from 'nock';
 import { success } from './responses';
 import 'isomorphic-fetch';
 import { path, SUCCESS_MESSAGE, url, uuidPattern } from './constants';
-import { PluginType, LogLevel, BaseEvent, Status } from '@amplitude/analytics-types';
+import { LogLevel, BaseEvent } from '@amplitude/analytics-types';
```
Comment (kelsonpw) [inline packages/analytics-browser-test/test/index.test.ts:9]:
```suggestion
import type { LogLevel, BaseEvent } from '@amplitude/analytics-types';
```

**Review by kelsonpw — APPROVED** [review]

File: packages/analytics-types/src/plugin.ts [hunk 2]
```diff
@@ -3,30 +3,29 @@ import { Config } from './config';
 import { Result } from './result';
 import { CoreClient } from './client/core-client';
 
-export enum PluginType {
-  BEFORE = 'before',
-  ENRICHMENT = 'enrichment',
-  DESTINATION = 'destination',
+type BeforePluginType = 'before';
```
Comment (justin-fiedler) [inline packages/analytics-types/src/plugin.ts:6]: nit: I would prefix with `PluginType` for easier code completion for end users. e.g. `PluginTypeBefore`, `PluginTypeEnrichment` so you can do `Plugin` + ctrl + space and see all the options.

Comment (kevinpagtakhan) [reply packages/analytics-types/src/plugin.ts]: Fixed: https://github.com/amplitude/Amplitude-TypeScript/pull/407/commits/edfd8dc7bf733e2a7ef10e1e3b8fa2d3ed976551

File: packages/analytics-browser-test/test/index.test.ts (hunk 1 above)
Comment (kevinpagtakhan) [reply packages/analytics-browser-test/test/index.test.ts:9]: Thanks @kelsonpw. Is it ok if I address this in a later PR?

Comment (kelsonpw) [reply packages/analytics-browser-test/test/index.test.ts:9]: Yes, total nit and can be ignored, I just personally find `type` imports to help readability

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**